coffeecard/CoffeeCard.Library/Services/v2/ProductService.cs

```diff
@@ -57,6 +58,92 @@ public async Task<Product> GetProductAsync(int productId)
             return product;
         }
 
+        public async Task<ProductResponse> AddProduct(AddProductRequest newProduct, IEnumerable<UserGroup> allowedUserGroups)
+        {
+            var product = new Product()
```

Comment:
I think we should check for uniqueness, e.g. check that no such product name already exists to avoid duplicate product names